### ISSUE TITLE
a2p produce syntax error 

### ISSUE THREAD

**Issue by dominix** [Leont/app-a2p#2]
example of use:
`echo 'BEGIN{OFS=","}{print $3,$1}'|a2p -F:`

this produce 

~~~~
#!/usr/bin/perl
eval 'exec /usr/bin/perl -S $0 ${1+"$@"}'
    if $running_under_some_shell;
			# this emulates #! processing on NIH machines.
			# (remove #! line above if indigestible) 
eval '$'.$1.'$2;' while $ARGV[0] =~ /^([A-Za-z_0-9]+=)(.*)/ && shift;
			# process any FOO=bar switches 
$FS = ':';		# field separator from -F switch
$, = ' ';		# set output field separator
$\ = "\n";		# set output record separator
$, = ',';
hile (<>) {
     ($Fld1,$Fld2,$Fld3) = split(/[:\n]/, $_, -1);
    print $Fld3, $Fld1;
}
~~~~

remark the line  `hile (<>)` instead of `while (<>)`

**Comment by Leont** [comment]
> remark the line hile (<>) instead of while (<>)

It appears to be an off-by-one somewhere. A somewhat mysterious one so far.

>  sorry not able to reproduce formating

Putting the code in a block starting and ending with `~~~~`  should fix that ;-)

**Comment by dominix** [comment]
thanks, edited formatting :-)

**Comment by dominix** [comment]
if I replace the -F switch with a variable in BEGIN, I have another side effect

~~~~
echo 'BEGIN{OFS=",";RS=":"}{print $3,$1}'|a2p

#!/usr/bin/perl
eval 'exec /usr/bin/perl -S $0 ${1+"$@"}'
    if $running_under_some_shell;
			# this emulates #! processing on NIH machines.
			# (remove #! line above if indigestible)

eval '$'.$1.'$2;' while $ARGV[0] =~ /^([A-Za-z_0-9]+=)(.*)/ && shift;
			# process any FOO=bar switches

$, = ' ';		# set output field separator
$\ = "\n";		# set output record separator

$, = ',';
$/ = ':';

whie (<>) {
     ($Fld1,$Fld2,$Fld3) = split(' ', $_, -1);
    print $Fld3, $Fld1;
}
~~~~ 
remark the line `whie (<>)` instead of `while (<>)`

**Comment by Leont** [comment]
This was one of the most mysterious bugs I've ever encountered. After some very deep digging I've discovered that it contains a `strcpy` to overlapping strings, which causes undefined behavior. The solution is to [not use](https://github.com/Leont/app-a2p/commit/aed2fc9799cbb1bf03f35491468b55a4458273bd) `strcpy`.

Interestingly, this code is older than the C standard (1987 versus 1989), and it appears this wasn't considered undefined behavior when it was written (I haven't found any mention of this restriction in manpages earlier than 1993)

**Comment by Leont** [comment]
Released as App-a2p 1.011
